app/pages/dashboard/components/table.test.js

```diff
@@ -0,0 +1,103 @@
+import { createTestHarness } from '../../../test-utils/testHarness';
+
+const setup = {
+  component: {
+    name: 'table',
+    path: 'pages/dashboard/components/table.njk',
+  },
+};
+
+const mockContext = {
+  params: {
+    title: 'Manage organisations and users',
+    description: 'Edit existing account information or add new user to an organisation.',
+    columnInfo: [
+      { data: 'Organisation name', link: true },
+      { data: 'ODS code', link: false },
+    ],
+    columnClass: 'nhsuk-grid-column-one-half',
+    data: [['Greater Manchester CCG', 'X01'], ['Hampshire CCG', 'X02']],
+  },
+};
+
+describe('table', () => {
+  it('should render the table', createTestHarness(setup, (harness) => {
+    harness.request(mockContext, ($) => {
+      expect($('[data-test-id="table"]').length).toEqual(1);
+    });
+  }));
+
+  it('should render the table headings with correct classes if columnInfo is passed in', createTestHarness(setup, (harness) => {
+    harness.request(mockContext, ($) => {
+      expect($('[data-test-id="table-headings"]').length).toEqual(1);
+      expect($('[data-test-id="column-heading"]').length).toEqual(mockContext.params.columnInfo.length);
+      mockContext.params.columnInfo.forEach((heading, i) => {
+        expect($(`[data-test-id="column-heading"]:nth-child(${i + 1})`).text().trim()).toEqual(heading.data);
+        expect($(`[data-test-id="column-heading"]:nth-child(${i + 1})`).hasClass(mockContext.params.columnClass)).toEqual(true);
+      });
+    });
+  }));
+
+  it('should not render the table headings if no data is passed in', createTestHarness(setup, (harness) => {
+    const context = { params: { ...mockContext.params } };
```

Comment:
is this just so we take a copy of mockContext and can then delete something from it?

Reply:
yes

Reply:
In that case can we use Object.assign() instead?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

Reply:
they do the same thing

Reply:
`it will look like this: const context = { params: Object.assign({}, mockContext.params) };` since both of them do a shallow copy

Reply:
yikes that looks even worse.